src/fullscreen-image-carousel/fullscreen-image-carousel.tsx

```diff
@@ -214,49 +261,75 @@ export const Component = (
     };
 
     return (
-        <Modal {...otherProps} data-testid="image-carousel-modal">
-            <CloseButton
-                aria-label="Close image carousel"
-                onClick={onClose}
-                focusHighlight={false}
-            >
-                <CrossIcon aria-hidden />
-            </CloseButton>
-            <ImageGalleryContainer>
-                <ImageGalleryWrapper>
-                    <ArrowButton
-                        aria-label="View previous image"
-                        data-testid="prev-btn"
-                        $position="left"
-                        onClick={goToPrevSlide}
-                    >
-                        <ChevronLeftIcon aria-hidden />
-                    </ArrowButton>
-                    <ArrowButton
-                        aria-label="View next image"
-                        data-testid="forward-btn"
-                        $position="right"
-                        onClick={goToNextSlide}
-                    >
-                        <ChevronRightIcon aria-hidden />
-                    </ArrowButton>
-                    <ImageGallerySwipe
-                        ref={containerRef}
-                        onTouchStart={handleTouchStart}
-                        onTouchMove={handleTouchMove}
-                        onTouchEnd={handleTouchEnd}
+        <FullscreenImageCarouselContext.Provider
+            value={{
+                state: fullscreenImageCarouselState,
+                saveImageDimension,
+            }}
+        >
+            <Modal {...otherProps} data-testid="image-carousel-modal">
+                <CloseButton
+                    aria-label="Close image carousel"
+                    onClick={onClose}
+                    focusHighlight={false}
+                >
+                    <CrossIcon aria-hidden />
+                </CloseButton>
+                {!hideMagnifier && (
+                    <MagnifierButton
+                        aria-label="Magnify image"
```

Comment:
for UX to advise on the copy, but could the button aria label be `Zoom in` and `Zoom out` depending on the current zoom?